stout/take.h

```diff
@@ -0,0 +1,191 @@
+#include <vector>
+
+#include "stout/eventual.h"
+#include "stout/filter.h"
+#include "stout/stream.h"
+
+namespace stout {
+namespace eventuals {
+namespace detail {
+
+////////////////////////////////////////////////////////////////////////
+
+struct _TakeLastN {
+  template <typename K_, typename Arg_>
+  struct Continuation : public TypeErasedStream {
+    // NOTE: explicit constructor because inheriting 'TypeErasedStream'.
+    Continuation(K_ k, size_t n)
+      : k_(std::move(k)),
+        N_(n) {}
+
+    void Start(TypeErasedStream& stream) {
+      stream_ = &stream;
+      data_ = std::vector<int>(N_);
+      k_.Start(*this);
+    }
+
+    template <typename... Args>
+    void Fail(Args&&... args) {
+      k_.Fail(std::forward<Args>(args)...);
+    }
+
+    void Stop() {
+      k_.Stop();
+    }
+
+    template <typename... Args>
+    void Body(Args&&... args) {
+      if (i_ == data_.size()) {
+        i_ = 0;
+        filled_ = true;
+      }
+      if (filled_) {
```

Comment:
Do we need to store a variable for this, or can we also just do `data_.size() == n_`?

If you think it's better for readability you can also consider a function named `Filled`:
```c++
bool Filled() {
  return data_.size() == n_;
}
```

Keeping track of state in multiple different variables is error-prone, hence my desire to avoid it.